### PR TITLE
Added printer

### PR DESCRIPTION
**Pull request by joshepvodka**
## About the PR

Printers now exist! Yay!
It's an old printer that is very unreliable and gets the paper stuck in it's rollers often.
It has a 1 in 20 chance to get stuck every time you try to print something, requiring you to pull the paper out.
If power goes out or you try to touch the paper tray while its printing it will also get stuck.
HoP will have a real good time while engis replace the ame fuel.

## Why / Balance

Make it easier to get your pre-made documents into the game. Less copy pasting.

## Technical details

The printer works with a 5 slot storage to act as the paper tray. Upon printing it drops the item, if the storage is empty then it doesn't print.
I left in SharedPrinterSystem even though it isn't working so someone can give me a hand on how to make it work. Basically i want to send popups to the player when they try to remove or add more paper into the printer's inventory while it is in a stuck state. Just for flavor basically, not necessary.

Also, I'm really bad at choosing a sound for the printer, so for now it has none. If anyone can help me out i'd appreciate it.

2nd also, they need to be mapped. If someone is so kind to map them for me they will get many gifts from santa.

## Media

https://github.com/space-wizards/space-station-14/assets/86210200/45cbc500-43af-40c8-816b-1d8307f6af65

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- add: NanoTrasen found a couple old printers in a warehouse.

### CODE REVIEW

**Comment by JoeHammad1844** [comment]
can you photocopy your ass?

**Comment by lzk228** [comment]
so you need to have form on your pc? thats weird and can be done with pen
photocopier would be better, it can print form from list and copy exist forms

**Comment by joshepvodka** [comment]
> so you need to have form on your pc? thats weird and can be done with pen photocopier would be better, it can print form from list and copy exist forms

So you're saying that it isn't a good addition to the game?
I think its fun as a roleplay item. Also, as someone that likes doing a lot of paperwork every shift I find it really annoying to have to alt tab, open a text file, copy everything, alt tab back and paste it on a sheet of paper. Having an item with a functionality like this would make us pen-pusher's lives so much easier.

It's true that it cant copy things, I might add a scan feature to it.

**Comment by LankLTE** [comment]
The problem of needing to make a lot of papers is better solved by #22027 imo, as that doesn’t require yet more machines to be mapped.

**Comment by Hmeister-fake** [comment]
> The problem of needing to make a lot of papers is better solved by #22027 imo, as that doesn’t require yet more machines to be mapped.

Agreed, no need to clog up space with printers. They are a cool setpiece though, so maybe add them anyway just don't map them.

**Comment by Simyon264** [comment]
> so you need to have form on your pc? thats weird and can be done with pen photocopier would be better, it can print form from list and copy exist forms

I've had issues where newlines were not pasted in properly and this would solve that. But I agree, a whole new machine is not needed for this. Maybe this functionality can be added to the fax machine instead?

**Comment by joshepvodka** [comment]
> > so you need to have form on your pc? thats weird and can be done with pen photocopier would be better, it can print form from list and copy exist forms
> 
> I've had issues where newlines were not pasted in properly and this would solve that. But I agree, a whole new machine is not needed for this. Maybe this functionality can be added to the fax machine instead?

Yeah I think it would now make sense to add this to faxes, seeing the fax copier PR..
But also maybe add printers to places that don't have any faxes, so people that aren't heads have easy access to bureaucracy, such as cargo and security.

**Comment by IamVelcroboy** [comment]
100% if you sit on it, it should send an ascii image of a bum.

**Comment by Emisse** [comment]
this should be merged with the fax machine so we dont need two separate machines for similar functions

**Comment by joshepvodka** [comment]
> this should be merged with the fax machine so we dont need two separate machines for similar functions

If i do could more fax machines be mapped in or a fax machine board be introduced to the game?

**Comment by Emisse** [comment]
yea i can map more fax machines in

File: Content.Client/Printer/UI/PrinterWindow.xaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+            <LineEdit Name="Name" MinSize="60 0" />
+        </BoxContainer>
+        <BoxContainer Orientation="Horizontal" HorizontalExpand="True">
+            <Label Text="Number of Copies:"/>
+        </BoxContainer>
+        <BoxContainer Orientation="Horizontal" HorizontalExpand="True">
+            <SpinBox Name="CopiesNumber" MinSize="60 0" />
+        </BoxContainer>
+        <BoxContainer Orientation="Horizontal" HorizontalExpand="True">
+            <Button Name="FileButton" Text="Print File" TextAlign="Center"/>
```
Comment (ficcialfaint) [inline Content.Client/Printer/UI/PrinterWindow.xaml:18]: This should be localized

File: Content.Server/Printer/PrinterSystem.cs [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+        component.FileText = args.Content;
+        component.FileName = args.Name;
+        component.FilesRemaining = args.Number;
+        if(component.PrintingTimeLeft == 0f)
+            StartPrinting(uid, component);
+    }
+
+    private void StartPrinting(EntityUid uid, PrinterComponent component)
+    {
+        if(_random.Prob(0.05f))
```
Comment (ficcialfaint) [inline Content.Server/Printer/PrinterSystem.cs:121]: Make the paper stuck chance a datafield please

File: Content.Server/Printer/PrinterSystem.cs [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+            user,
+            TimeSpan.FromSeconds(3),
+            new PrinterUnstuckEvent(),
+            uid
+        );
+        doAfterArgs.BreakOnUserMove = true;
+        doAfterArgs.NeedHand = true;
+        
+        _doAfterSystem.TryStartDoAfter(doAfterArgs);
+        _popupSystem.PopupEntity("You try to remove the stuck paper!", uid, user);
```
Comment (ficcialfaint) [inline Content.Server/Printer/PrinterSystem.cs:144]: Localization & for other popups and stuff

**Review by ficcialfaint — CHANGES_REQUESTED** [review]

**Comment by joshepvodka** [comment]
gonna merge this with the fax machine

**Pull request closed without merging**